be/src/storage/column_predicate_rewriter.cpp

```diff
@@ -48,46 +48,121 @@ namespace starrocks {
 
 constexpr static const LogicalType kDictCodeType = TYPE_INT;
 
-Status ColumnPredicateRewriter::rewrite_predicate(ObjectPool* pool) {
-    // because schema has reordered
-    // so we only need to check the first `predicate_column_size` fields
-    for (size_t i = 0; i < _column_size; i++) {
-        const FieldPtr& field = _schema.field(i);
-        const auto cid = field->id();
+struct RewritePredicateTreeVisitor {
+    using RewriteStatus = ColumnPredicateRewriter::RewriteStatus;
+
+    template <CompoundNodeType ParentType>
+    StatusOr<RewriteStatus> operator()(PredicateColumnNode& node, PredicateCompoundNode<ParentType>& parent) const {
+        const auto* col_pred = node.col_pred();
+        const auto cid = col_pred->column_id();
 
-        auto iter = _pred_map.find(cid);
-        if (iter == _pred_map.end()) {
-            continue;
+        if (!_rewriter._need_rewrite[cid]) {
```

Comment:
suggest to use a function instead of accessing to field directly.

Reply:
`RewritePredicateTreeVisitor` is just a part implementation of `ColumnPredicateRewriter::rewrite_predicate`. This is similar to IMPL. Therefore, I think it is ok to access private members of `ColumnPredicateRewriter`.